compiler/syntax/src/res_core.ml

```diff
@@ -268,7 +268,7 @@ let rec go_to_closing closing_token state =
 (* Madness *)
 let is_es6_arrow_expression ~in_ternary p =
   Parser.lookahead p (fun state ->
-      let async =
+      let _async =
```

Comment:
I wonder why we didn't get a warning before if this is unused?

Reply:
It was used for recovery on tilde token, I forgot to rollback the recovery change

Reply:
Maybe not. we should remove the tilide handling because now we allow to use it somewhere in that context.